src/core/layout/qgslayoutpagecollection.cpp

```diff
@@ -69,13 +71,48 @@ void QgsLayoutPageCollection::reflow()
 double QgsLayoutPageCollection::maximumPageWidth() const
 {
   double maxWidth = 0;
-  Q_FOREACH ( QgsLayoutItemPage *page, mPages )
+  for ( QgsLayoutItemPage *page : mPages )
   {
     maxWidth = std::max( maxWidth, mLayout->convertToLayoutUnits( page->pageSize() ).width() );
   }
   return maxWidth;
 }
 
+QSizeF QgsLayoutPageCollection::maximumPageSize() const
+{
+  double maxArea = 0;
+  QSizeF maxSize;
+  for ( QgsLayoutItemPage *page : mPages )
```

Comment:
Why not `qgis::as_const( mPages )` in this case? Sorry if it's a dumb question...

Reply:
It's already const here, because the method it's contained in is const.

Reply:
Indeed :)